### ISSUE TITLE
>=7.4.0: New bug: Missing classes for core/group block

### ISSUE THREAD

**Issue by strarsis** [WordPress/gutenberg#20347]
**Describe the bug**
Since release `7.4.0` of Gutenberg plugin (so `7.5.0` is also included) the DOM element of `core/group` block doesn't have classes anymore, notably for the selected block style (`is-style-<block style name>`. The classes are still correctly updated in the `Advanced`/`Additional CSS Class(es)` field.

**To reproduce**
1. Install/update Gutenberg plugin `7.4.0` or higher (`7.5.0`).
2. Open Gutenberg page editor, ensure the theme offers at least one non-default block style for the `core/group` block to be selected.
3. Insert a new `core/group` block, to ensure no other markup is re-used from previous, correctly working Gutenberg releases, in order to reproduce this bug correctly.
4. Select a block style for the `core/group` block.
5. Inspect the element and notice that the DOM element for `core/group` blocks (with `wp-block-group` class) don't have any other classes assigned.
6. Expand the `Advanced` pane, notice that the `is-style-*` class is correctly listed in the `Additional CSS Class(es)` field for that block.

**Expected behavior**
The classes should be applied to the DOM element of the `core/group` block as in releases of Gutenberg prior to `7.4.0`.

**Desktop (please complete the following information):**
 - OS: Windows 10 Pro x64
 - Browser: Chrome
 - Version: Latest

**Additional context**
When the Gutenberg plugin is disabled (only WordPress core used) or downgraded to a release prior to `7.4.0`, like `7.3.0`, the issue goes away.

**Comment by jorgefilipecosta** [comment]
Hi @strarsis,
I'm not being able to reproduce this issue.
I added the following code to the functions.php file of the currently enabled theme:
```
register_block_style(
    'core/group',
    array(
        'name'         => 'blue-group',
        'label'        => __( 'Blue Group' ),
        'inline_style' => '.wp-block-group.is-style-blue-group { background-color: blue; }',
    )
);
````

I verified a Blue Group style was available and I verified that the class is-style-blue-group was applied to the dom of the editor and to the dom in the frontend. The block gained a blue background as expected.


Could you provide additional details about your test website: plugins installed, the theme used, or any other information you may find relevant.
Thank you in advance!

**Comment by strarsis** [comment]
@jorgefilipecosta: This I used for registering a block style in theme on backend:
```php
add_action( 'init', function() {
    if(!function_exists('register_block_style')) return;

    // Block styles

    // Group
    register_block_style('core/group', [
        'name'  => 'test',
        'label' => 'Test',
    ]);

});
````

**Comment by talldan** [comment]
Hey @strarsis, thanks for reporting. This one was reported in https://github.com/WordPress/gutenberg/issues/20140 and fixed by https://github.com/WordPress/gutenberg/pull/20164. The fix will be in Gutenberg 7.6, which will be released this week.

**Comment by strarsis** [comment]
@talldan: I am really looking forward to the next Gutenberg release! 🐱
